service/controller/v25/detection/detection.go

```diff
@@ -0,0 +1,92 @@
+package detection
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/aws-operator/service/controller/v25/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v25/key"
+)
+
+type Config struct {
+	Logger micrologger.Logger
+}
+
+// Detection is a service implementation deciding if a tenant cluster should be
+// updated or scaled.
+type Detection struct {
```

Comment:
Here did the implementation move to. The rules are still the same, I just polished the code a bit further.